k2/csrc/context.h

```diff
@@ -550,15 +550,33 @@ class ParallelRunner {
   // will call CudaStreamOverride::OverrideStream() and replace it
   // with this stream automatically.
   cudaStream_t NewStream() {
-    // TODO
-    return kCudaStreamInvalid;
+    DeviceType d = c_->GetDeviceType();
+    if (d == kCpu) {
+      return kCudaStreamInvalid;
+    } else {
+      K2_CHECK_EQ(d, kCuda);
+      cudaStream_t stream;
+      auto ret = cudaStreamCreate(&stream);
```

Comment:
after creating the stream we need to record an event in the stream c_.Stream(), and have stream `stream` wait on that event.